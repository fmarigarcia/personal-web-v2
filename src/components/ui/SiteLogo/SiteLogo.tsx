import React from 'react';
import clsx from 'clsx';
import { Logo } from '@icons/Logo';
import { useTheme } from '@contexts/ThemeContext';
import { LOGO_COLORS } from '@utils/constants';

interface LogoProps {
    onClick?: () => void;
    className?: string;
}

export const SiteLogo: React.FC<LogoProps> = ({ onClick, className = '' }) => {
    const {
        data: { theme },
    } = useTheme();
    return (
        <div className={clsx('flex items-center', className)}>
            <button
                onClick={onClick}
                className="text-xl font-bold text-stone-800 dark:text-green-400 hover:text-stone-950 dark:hover:text-green-300 transition-colors"
                type="button"
                aria-label="Go to home"
            >
                <Logo fill={LOGO_COLORS[theme]} />
            </button>
        </div>
    );
};

export default SiteLogo;
