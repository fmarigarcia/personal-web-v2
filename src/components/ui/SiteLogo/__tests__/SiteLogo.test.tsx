import { render, screen, fireEvent } from '@testing-library/react';
import { SiteLogo } from '../SiteLogo';

// Mock the Logo component from icons
jest.mock('@icons/Logo', () => ({
    Logo: ({ fill }: { fill?: string }) => (
        <svg data-testid="logo-svg" width="32" height="32">
            <path fill={fill} />
        </svg>
    ),
}));

// Mock the theme context
jest.mock('@contexts/ThemeContext', () => ({
    useTheme: () => ({
        data: { theme: 'light' },
    }),
}));

// Mock the constants
jest.mock('@utils/constants', () => ({
    LOGO_COLORS: {
        light: '#000',
        dark: '#fff',
    },
}));

describe('SiteLogo', () => {
    it('should render logo with SVG', () => {
        render(<SiteLogo />);

        const button = screen.getByRole('button', { name: /go to home/i });
        const svg = screen.getByTestId('logo-svg');

        expect(button).toBeInTheDocument();
        expect(svg).toBeInTheDocument();
    });

    it('should handle click when onClick prop is provided', () => {
        const mockOnClick = jest.fn();
        render(<SiteLogo onClick={mockOnClick} />);

        const button = screen.getByRole('button');
        fireEvent.click(button);

        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });

    it('should not crash when clicked without onClick prop', () => {
        render(<SiteLogo />);

        const button = screen.getByRole('button');
        expect(() => fireEvent.click(button)).not.toThrow();
    });

    it('should apply custom className', () => {
        const customClass = 'custom-logo';
        render(<SiteLogo className={customClass} />);

        const logoContainer = screen.getByRole('button').parentElement;
        expect(logoContainer).toHaveClass(customClass);
    });

    it('should have proper accessibility attributes', () => {
        render(<SiteLogo />);

        const button = screen.getByRole('button');
        expect(button).toHaveAttribute('type', 'button');
        expect(button).toHaveAttribute('aria-label', 'Go to home');
    });

    it('should have hover styles', () => {
        render(<SiteLogo />);

        const button = screen.getByRole('button');
        expect(button).toHaveClass('hover:text-stone-950', 'transition-colors');
    });

    it('should match snapshot', () => {
        const { container } = render(<SiteLogo />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
