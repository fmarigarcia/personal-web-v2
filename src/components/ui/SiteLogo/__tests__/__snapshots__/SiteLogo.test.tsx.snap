// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteLogo should match snapshot 1`] = `
<div
  class="flex items-center"
>
  <button
    aria-label="Go to home"
    class="text-xl font-bold text-stone-800 dark:text-green-400 hover:text-stone-950 dark:hover:text-green-300 transition-colors"
    type="button"
  >
    <svg
      data-testid="logo-svg"
      height="32"
      width="32"
    >
      <path
        fill="#000"
      />
    </svg>
  </button>
</div>
`;
